apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -85,6 +95,31 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
     }
   }, [currentAiModalField]);
 
+  useEffect(() => {
```

Comment:
Would it be better to just compute and cache these translations ahead of time? The content of the block dropdowns isn't going to change when the results are calculated right?

Reply:
Yep, I think you're right. Trying out `useMemo` here for the first time, thoughts?

https://github.com/code-dot-org/code-dot-org/pull/54534/commits/b1a7d6be9b38ff5a7a5ea2dd8134229d6eea715f

Reply:
Yep, seems reasonable!